### ISSUE TITLE
Replace FMDB with CoreData

### ISSUE THREAD

**Issue by amjaliks** [route-me/route-me#77]
Replace FMDB with CoreData.

**Comment by andrewljohnson** [comment]
You don't want to replace FMBD with CoreData. You want to replace FMDB with a file-system-based solution. It's much, much faster and uses less memory for all operations, including mass-caching tiles, displaying tiles, and deleting tiles.

**Comment by amjaliks** [comment]
Agree, for tile storage FS may be preferable. But still need some backend to store metadata about cache contents.

**Comment by incanus** [comment]
Yeah, agreed on replacing with Core Data. Way more dependencies, much more memory overhead, and all we're doing is caching, not maintaining an object graph. 

Incidentally, the MBTIles backend makes use of FMDB as well, since the entire file format is SQLite-based and meant to be portable to other (including non-Cocoa) systems.
